library_builders.bat

```diff
@@ -161,6 +161,11 @@ exit /b 0
   cmake --build build-zlib --target install --config Release
   if errorlevel 1 exit 1
 
+  set "zlib_dll=%BUILD_PREFIX:~1,-1%/zlib/bin/zlib.dll"
+  set "zlib_dll=%zlib_dll:/=\%"
+  del "%zlib_dll%"
```

Comment:
Windows builds still pick up a `zlib.dll` somewhere, probably from the system...

Reply:
HDF5:
```
2023-03-28T19:36:25.6925189Z -- Could NOT find ZLIB (missing: ZLIB_DIR)
2023-03-28T19:36:25.8188114Z -- Found ZLIB: C:/Program Files (x86)/zlib/lib/zlib.lib (found version "1.2.12")  
```

Reply:
Compiled without any backends, the `.dll` dependencies are reduced to only allowed ones:
```
$ objdump -p openpmd_api_cxx.cp311-win_amd64.pyd | grep dll
	DLL Name: python311.dll
	DLL Name: KERNEL32.dll
	DLL Name: MSVCP140.dll
	DLL Name: VCRUNTIME140_1.dll
	DLL Name: VCRUNTIME140.dll
	DLL Name: api-ms-win-crt-string-l1-1-0.dll
	DLL Name: api-ms-win-crt-heap-l1-1-0.dll
	DLL Name: api-ms-win-crt-runtime-l1-1-0.dll
	211e44	   63  _seh_filter_dll
	DLL Name: api-ms-win-crt-convert-l1-1-0.dll
	DLL Name: api-ms-win-crt-math-l1-1-0.dll
	DLL Name: api-ms-win-crt-stdio-l1-1-0.dll
	DLL Name: api-ms-win-crt-locale-l1-1-0.dll
	DLL Name: api-ms-win-crt-filesystem-l1-1-0.dll
	DLL Name: api-ms-win-crt-environment-l1-1-0.dll
	DLL Name: api-ms-win-crt-time-l1-1-0.dll
```

Reply:
Setting the HDF5 hints to our zlib explicitly does not help.